datafusion/core/src/datasource/listing/table.rs

```diff
@@ -548,7 +549,7 @@ impl ListingTable {
         })?;
 
         // Add the partition columns to the file schema
-        let mut builder = SchemaBuilder::from(file_schema.fields());
+        let mut builder = SchemaBuilder::from(file_schema.deref().clone());
```

Comment:
```suggestion
        let mut builder = SchemaBuilder::from(file_schema.as_ref().to_owned());
```

nit: I think this is achieves the same but doesn't require importing `Deref`

Reply:
You are right, I would make the change.